ops/get_upstream.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+#set -eo pipefail
+source ops/common.sh
+
+function buildUpstream() {
+  name=$(echo -n $1 | cut -f2 -d/) 
+  container=$1
+  gitHash=$2
+  tag=$3
+  repoPath=/tmp/$name
+
+  if [[ "$name" == "authority" ]]; then
+    # using nonvoting authority
+    repoUrl=$katzenAuthRepo
+    dockerFile=$repoPath/Dockerfile.nonvoting
+    old="RUN cd cmd/nonvoting \&\& go build"
+    new="RUN cd cmd/nonvoting \&\& go build$warpedBuildFlags"
+  elif [[ "$name" == "server" ]]; then
+    repoUrl=$katzenServerRepo
+    dockerFile=$repoPath/Dockerfile
+    old="RUN cd cmd/server \&\& go build"
+    new="RUN cd cmd/server \&\& go build$warpedBuildFlags"
+  else
+    LOG "Variable \$name is not either authority or server. Stopping"
+    exit 1
+  fi
+
+  LOG "Building upstream... $container:$tag"
+  rm -rf $repoPath && git clone $repoUrl $repoPath > /dev/null
+  cd $repoPath > /dev/null
+  git -c advice.detachedHead="false" checkout $gitHash > /dev/null
+  sed -i "s|$old*|$new|" $dockerFile
```

Comment:
Instead of finding an replacing, you can just find the old string and appended the whatever `$wardpedBuildFlags` is at the end. It makes the code much simpler and you can omit of lines 12 to 23.

Reply:
I still need an if to find out which Dockerfile I'm going to append to. Auhtority has a different dockerfile to server.

Reply:
I meant you can omit most of lines 12 to 13 i.e. the if statements are simplified.

Reply:
I agree that can be done. I spent some time trying to solve that but have not been able to come with a solution.